decoder/s2cell.go

```diff
@@ -74,3 +75,35 @@ func saveS2CellRecords(ctx context.Context, db db.DbDetails, cellIds []uint64) {
 		s2CellCache.Set(cellId.Id, cellId, ttlcache.DefaultTTL)
 	}
 }
+
+type EmptyCellTracker struct {
+	Data *xsync.MapOf[uint64, int]
+}
+
+func NewEmptyCellTracker() *EmptyCellTracker {
+	return &EmptyCellTracker{
+		Data: xsync.NewIntegerMapOf[uint64, int](),
+	}
+}
+
+func (tracker *EmptyCellTracker) IncreaseCount(cellId uint64) {
+	tracker.Data.Compute(cellId, func(oldValue int, loaded bool) (newValue int, delete bool) {
+		if loaded {
+			return oldValue + 1, false
+		} else {
+			return 1, false
+		}
+	})
+}
+
+func (tracker *EmptyCellTracker) ResetCount(cellId uint64) {
+	tracker.Data.Delete(cellId)
+}
+
+func (tracker *EmptyCellTracker) ShouldConsiderEmpty(cellId uint64) bool {
+	count, exists := tracker.Data.Load(cellId)
+	if !exists {
+		return false
+	}
+	return count >= 5
```

Comment:
As I said before, I am concerned that you will consistently get empty cells at the edge of eg 9x9 when outside of max radius; then along comes another type of scan (eg quests vs raid) and now it is suddenly full.
I am not sure how you diagnose this permanently instead of guessing at numbers though (unless you never do a clear when a cell is empty)

Reply:
This is quite impossible to handle in Golbat, we would have to consider the scanContext, that's not used by any public controller :/

In this case empty really means -> no forts and no mons.
It can also be that a cell does not have any content in one RAW, but in the next there is -> this emptyCell check is more for that reasons to prevent processing empty GMOs

Reply:
If this eliminates false removes, fine -- but I suspect it won't and we'll have to move to a time based approach to catch the 1->0 removals reliably. It's worth a go but since you've increased it I imagine it's not actually working